sw/host/tests/chip/i2c_target/src/main.rs

```diff
@@ -135,6 +141,50 @@ fn test_wakeup_deep_sleep(
     test_read_transaction(opts, transport, address)
 }
 
+fn test_repeated_start(_opts: &Opts, transport: &TransportWrapper, address: u8) -> Result<()> {
+    let uart = transport.uart("console")?;
+
+    log::info!("Emulating start with bit-banging");
+    let gpio_bitbanging = transport.gpio_bitbanging()?;
+    let gpio_pins = transport.gpio_pins(&["IOA7", "IOA8"].map(|s| s.to_string()))?;
+    let i2c_bitbang = test_utils::bitbanging::i2c::Host::<0, 1> {};
+
+    for pin in &gpio_pins {
+        pin.set_mode(PinMode::OpenDrain)?;
+    }
+    const REFERENCE_DATA: &[u8] = b"Hello World!";
+    let transaction = &i2c_bitbang.build_transaction(&[
```

Comment:
I have not finished reviewing the implementation of the transaction builder but more generally on the transaction itself, I think it would be interesting to test a more realistic transaction with repeated start which is one where one first does a write (to e.g. set a register address) and then repeated start with a read. The point is that the direction of the transfer is different.
What do you think?

Reply:
This is not what I understood from the testplan:
           - Program the I2C to be in target mode.
            - Start a transaction, but fail to send a stop.
            - Send a repeated start with a specific data value and stop the transaction as usual.

Reply:
Ok fair enough. But then that does not seem like a very useful/realistic test, repeated starts are never used like this.

Reply:
"Start a transaction" can be interpreted as "Sending a start and a data value". That's how I would interpret it.

Reply:
The "...but _fail_ to send a stop" case is particularly interesting in a multi-host context, where a multi-host-capable i2c IP would detect bus idle and take over with its own transaction.

However, the most interesting case for this single-host IP would be to do a write command followed by a read command in a single transaction. This is the most common use case of repeated starts.

Btw, in i2c parlance, a "transaction" is the full reserved bus time from the first Start to the Stop, and it may encompass multiple commands/transfers (with each transfer beginning with a Start). In the test plan, we should probably take care with how we use the terms.

Reply:
I added another test routine to verify the write and read in the same transaction.